### PR TITLE
update roadmap for v0.9+

### PR DESCRIPTION
**Pull request by jbw976**
**Description of your changes:** This PR updates the Rook roadmap for v0.9 and future milestones

**Checklist:**
- [x] Documentation has been updated, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [x] `make vendor` does not cause changes.

[skip ci]

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -1,40 +1,64 @@
...
+* Integrate a more robust controller framework (e.g., CoreOS Operator SDK or Kubebuilder) [#1981](https://github.com/rook/rook/issues/1981)
+* Build and integration testing improvements
+  * Increase PR quality gates (e.g., [vendoring verification](https://github.com/rook/rook/issues/1822), [license scanning](https://github.com/rook/rook/issues/1993), etc.)
+  * Update promotion and release channels to align with storage provider specific statuses [#1885](https://github.com/rook/rook/issues/1885)
+  * Refactor test framework and helpers to support multiple storage providers [#1788](https://github.com/rook/rook/issues/1788)
+  * Isolate and parallelize storage provider testing [#1218](https://github.com/rook/rook/issues/1218)
+  * Longhaul testing pipeline [#1847](https://github.com/rook/rook/issues/1847)
+* Custom resource validation, progress, status [#1539](https://github.com/rook/rook/issues/1539)
+* Design for Volume Snapshotting and policies (consider aligning with SIG-storage) [#1552](https://github.com/rook/rook/issues/1552)
+* Support for Kubernetes 1.9+
```
Comment (jbw976) [inline ROADMAP.md]: @travisn I saw this item in your notes, but I'm not sure what the specific work items or issues here are. Can you elaborate?

File: ROADMAP.md [hunk 2]
```diff
@@ -1,40 +1,64 @@
...
-* Remove Rook API and CLI
-* Run with Least Privileged design
-  * Improved support for OpenShift
-* Support Kubernetes 1.7+ only
-* Ceph features and improvements
-  * Disk management (1 OSD per pod, adding/removing disks)
-  * Placement group balancer support (ceph-mgr module)
-  * Mon reliability (one mon per node, failing over too fast, etc.)
-
 ## Rook 0.9
```
Comment (jbw976) [inline ROADMAP.md]: There are a LOT of items in the 0.9 milestone, I am very open to shifting some of them to future milestones.  We can also consider a 0.10 milestone before we get to 1.0.

File: ROADMAP.md [hunk 3]
```diff
@@ -1,40 +1,64 @@
...
 ## Rook 1.0
 
-* Declare numerous types as v1
-* Automated upgrades
+* Declare Ceph to be stable v1
+* Durability of state (local storage support, config can be regenerated) [#1011](https://github.com/rook/rook/issues/1011) [#592](https://github.com/rook/rook/issues/592)
+* Integration testing improvements
+  * Incorporate new environments [#1315](https://github.com/rook/rook/issues/1315)
+  * Incorporate more architectures [#1901](https://github.com/rook/rook/issues/1901)
+* Support for dynamic provisioning of new storage types
```
Comment (bassam) [inline ROADMAP.md]: I'd like to see these move up to v0.9 if we can find an owner.

File: ROADMAP.md [hunk 4]
```diff
@@ -1,40 +1,64 @@
...
+  * Dynamic bucket provisioning [#1705](https://github.com/rook/rook/issues/1705)
+  * Dynamic database provisioning [1704](https://github.com/rook/rook/issues/1704)
+* New storage providers
+  * Cassandra cluster creation and horizontal scaling [#1910](https://github.com/rook/rook/issues/1910)
+  * Nexenta [#1532](https://github.com/rook/rook/issues/1532)
+* Ceph
+  * CSI plug-in (rook-ceph-agent uses ceph-fuse, nbd-rbd / tcmu runner) [#1385](https://github.com/rook/rook/issues/1385)
```
Comment (galexrt) [inline ROADMAP.md]: Isn't `plug-in` normally written as `plugin`?

File: ROADMAP.md (hunk 2 above)
Comment (travisn) [reply ROADMAP.md]: There are many items in this list that don't have owners yet, which puts them at risk for this milestone. This makes me wonder if there is a better way to represent the roadmap. The key is that we still reserve the right to add/remove items to the list and it is heavily dependent on the amount of community involvement. What if we added a comment to that effect at the top of the roadmap? I'd suggest we optimistically keep items in the 0.9 milestone until we get closer to the release date before we punt them to the next milestone.

File: ROADMAP.md (hunk 4 above)
Comment (jbw976) [reply ROADMAP.md]: they are both acceptable ways to spell it :) https://en.wikipedia.org/wiki/Plug-in_(computing)

File: ROADMAP.md (hunk 2 above)
Comment (jbw976) [reply ROADMAP.md]: i think this makes sense, i will add some language around this and we'll review.  this fits into @bassam's suggestion about moving some items up if we can find an owner for them.  perhaps this roadmap can help to serve as a "call for contributors" to take ownership for some of these features, with the caveat that some items may be pushed to later milestones if owners can't be found.  i'll take a stab at that.

**Comment by jbw976** [comment]
@travisn @bassam I have included language about the optimistic nature of the roadmap, how community engagement is important, and that final features included in a release is subject to change.  I've also moved dynamic provisioning of new storage types to the 0.9 milestone, hoping we will be able to find an owner.  PTAL

File: ROADMAP.md [hunk 5]
```diff
@@ -1,40 +1,67 @@
...
+* Minio
+  * Helm chart deployment [#1814](https://github.com/rook/rook/issues/1814)
+  * Run on arbitrary PVs [#919](https://github.com/rook/rook/issues/919)
+* Ceph
+  * Update to Mimic [#1536](https://github.com/rook/rook/issues/1536)
+  * Automated upgrade support (initial) [#997](https://github.com/rook/rook/issues/997)
+  * Manage an existing Ceph cluster (basic) [#1868](https://github.com/rook/rook/issues/1868)
+  * OSDs
+    * Run on arbitrary PVs (local storage) as an alternative to host path [#796](https://github.com/rook/rook/issues/796) [#919](https://github.com/rook/rook/issues/919)
+    * Minimize or eliminate running with privileged containers (e.g., reduced access to /dev) [#1944](https://github.com/rook/rook/issues/1944)
```
Comment (galexrt) [inline ROADMAP.md]: There is a comma too much after the `e.g.`

File: ROADMAP.md [hunk 6]
```diff
@@ -1,40 +1,67 @@
...
+    * Dynamic Volume Provisioning for CephFS [#1125](https://github.com/rook/rook/issues/1125)
+  * Object
+    * Multi-site configuration [#1584](https://github.com/rook/rook/issues/1584)
+    * CRD for object store users [#1583](https://github.com/rook/rook/issues/1583)
 
 ## Rook 1.0
 
-* Declare numerous types as v1
-* Automated upgrades
+* Declare Ceph to be stable v1
```
Comment (galexrt) [inline ROADMAP.md]: I think adding `CRDs` after Ceph would make it 100% clear what we are talking about.

File: ROADMAP.md [hunk 7]
```diff
@@ -1,40 +1,67 @@
...
+  * Run on arbitrary PVs [#919](https://github.com/rook/rook/issues/919)
+* Ceph
+  * Update to Mimic [#1536](https://github.com/rook/rook/issues/1536)
+  * Automated upgrade support (initial) [#997](https://github.com/rook/rook/issues/997)
+  * Manage an existing Ceph cluster (basic) [#1868](https://github.com/rook/rook/issues/1868)
+  * OSDs
+    * Run on arbitrary PVs (local storage) as an alternative to host path [#796](https://github.com/rook/rook/issues/796) [#919](https://github.com/rook/rook/issues/919)
+    * Minimize or eliminate running with privileged containers (e.g., reduced access to /dev) [#1944](https://github.com/rook/rook/issues/1944)
+    * Disk management (adding, removing, and replacing disks) [#1435](https://github.com/rook/rook/issues/1435)
+    * ceph-volume provisioning [#1342](https://github.com/rook/rook/issues/1342)
```
Comment (galexrt) [inline ROADMAP.md]: What about changing it to "`ceph-volume` is used for provisioning"?

File: ROADMAP.md (hunk 5 above)
Comment (jbw976) [reply ROADMAP.md]: I prefer the style of including a comma after i.e. and e.g..  From https://www.dailywritingtips.com/comma-after-i-e-and-e-g/:
> The Chicago Manual of Style states that i.e. and e.g. should be “confined to parentheses and notes and followed by a comma.”

> The AP Stylebook, whose “punctuation-pitch” leans generally to the side of “the fewer commas the better,” is pro-comma when it comes to i.e. and e.g. According to AP, both abbreviations are “always followed by a comma.”

**Comment by jbw976** [comment]
@galexrt incorporated your feedback (except the comma change), PTAL

**Pull request merged**